### DISCUSSION TITLE
Solution Importing...   Microsoft PowerPlatform CLI   Version: 1.25.2+g1c4a1b8   Online documentation: https://aka.ms/PowerPlatformCLI   Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions   Error: CanvasApp import: FAILURE: No file with that name found       CanvasApp import: FAILURE: No file with that name found   Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]

### DISCUSSION THREAD

**Discussion by bmhemanthreddy** [microsoft/powerplatform-build-tools#1056] [General]
Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]

**Comment by bmhemanthreddy** [comment]
facing issues while deploying the code to QA environment

**Reply by mannybellas** [reply]
Does this happen with the latest version of pac (1.35.1)?  1.25.2 is quite a few releases back

**Reply by bmhemanthreddy** [reply]
how to check the latest version of pac (1.35.1)

**Reply by mannybellas** [reply]
you can install the latest by `pac install latest`

**Reply by bmhemanthreddy** [reply]
where i need to pac install latest details in the below code

name: release-solution-to-prod-reusable
# Reusable workflow
# convert solution to managed (using a build PowerPlatform environment for the conversion)
# upload the solution to the GitHub artifacts and deploy to the PROD environment
on:
  workflow_call:
    inputs: 
      #Do Not change these values
      #Values are set by the caller
      #caller sample: release-action-call.ymnl
      solution_name:
        description: 'The solution name.'
        type: string
        default: ALMLab       
      solution_shipping_folder:
        description: 'folder name for staging the exported solution *do not change*'        
        type: string
        default: out/ship/
      solution_outbound_folder:
        description: 'staging the unpacked solution folder before check-in *do not change*'
        type: string
        default: out/solutions/
      solution_source_folder: 
       description: 'folder name to be created and checked in *do not change*'
       type: string
       default: solutions/
      solution_release_folder:
       description: 'folder where the released binaries are going to be hosted *do not change*'
       type: string
       default: out/release
      BUILD_ENVIRONMENT_URL:
        description: 'Build environment url.'
        type: string
        required: true      
      PRODUCTION_ENVIRONMENT_URL: 
        description: 'Production environment url.'
        type: string
        required: true
      CLIENT_ID: 
        description: 'The client id'
        type: string
        required: true
      TENANT_ID: 
        description: 'The tenant id'
        type: string
        required: true
    secrets:
      envSecret:
        description: 'The secret value for authentication using SPN'
        required: true

jobs:
  convert-to-managed:
    runs-on: windows-latest
    # or you can say runs-on: ubuntu-latest
    env:
      RUNNER_DEBUG: 1

    steps:
    - uses: actions/checkout@v2
      with:
        lfs: true

    - name: Pack solution
      uses: microsoft/powerplatform-actions/pack-solution@v0
      with:
        solution-folder: ${{ inputs.solution_source_folder}}/${{ inputs.solution_name }}
        solution-file: ${{ inputs.solution_outbound_folder}}/${{ inputs.solution_name }}.zip
        solution-type: Unmanaged

    - name: Import solution as unmanaged to build env
      uses: microsoft/powerplatform-actions/import-solution@v0
      with:
        environment-url: ${{inputs.BUILD_ENVIRONMENT_URL}}
        app-id: ${{inputs.CLIENT_ID}}
        client-secret: ${{ secrets.envSecret }}
        tenant-id: ${{inputs.TENANT_ID}}
        solution-file: ${{ inputs.solution_outbound_folder}}/${{ inputs.solution_name }}.zip
        force-overwrite: true
        publish-changes: true

    - name: Export solution as managed
      uses: microsoft/powerplatform-actions/export-solution@v0
      with:
        environment-url: ${{inputs.BUILD_ENVIRONMENT_URL}}
        app-id: ${{inputs.CLIENT_ID}}
        client-secret: ${{ secrets.envSecret }} 
        tenant-id: ${{inputs.TENANT_ID}}
        solution-name: ${{ inputs.solution_name }}
        managed: true
        solution-output-file: ${{ inputs.solution_shipping_folder}}/${{ inputs.solution_name }}_managed.zip

    - name: Upload the ready to ship solution to GH artifact store
      uses: actions/upload-artifact@v4
      with:
        name: managedSolutions
        path: ${{ inputs.solution_shipping_folder}}/

  release-to-staging:
    needs: [ convert-to-managed ]
    runs-on: windows-latest
    env:
      RUNNER_DEBUG: 1

    steps:
    - uses: actions/checkout@v2
      with:
        lfs: true

    - name: Fetch the ready to ship solution from GH artifact store
      uses: actions/download-artifact@v2
      with:
        name: managedSolutions
        path: ${{ inputs.solution_release_folder}}
    - name: Import solution to prod env
      uses: microsoft/powerplatform-actions/import-solution@v0
      with:
        environment-url: ${{inputs.PRODUCTION_ENVIRONMENT_URL}}
        app-id: ${{inputs.CLIENT_ID}}
        client-secret: ${{ secrets.envSecret }}
        tenant-id: ${{inputs.TENANT_ID}}
        solution-file: ${{ inputs.solution_release_folder}}/${{ inputs.solution_name }}_managed.zip
        force-overwrite: true
        publish-changes: true

**Reply by bmhemanthreddy** [reply]
error message:
Run microsoft/powerplatform-actions/import-solution@v0
loading action yaml file: D:\a\_actions\microsoft\powerplatform-actions\v0\import-solution\action.yml
import-solution:
  authN to env. authType:SPN authScheme:; cloudInstance: Public; envUrl: https://org0836b70a.crm4.dynamics.com/
  '5690f999-f149-4b9f-9e0e-528b0ab94cdb' authenticated successfully.
  Validating connection...
  Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f149-4b9f-9e0e-528b0ab94cdb
  Authentication profile created
      * DATAVERSE                                https://org0836b70a.crm4.dynamics.com/   : 5690f999-f149-4b9f-9e0e-528b0ab94cdb     Public
  
  The Authentication Result: '5690f999-f149-4b9f-9e0e-528b0ab94cdb' authenticated successfully.,Validating connection...,Connected to... EnterpriseAutomation_TCS_QA,Connected as 5690f999-f149-4b9f-9e0e-528b0ab94cdb,Authentication profile created,    * DATAVERSE                                https://org0836b70a.crm4.dynamics.com/   : 5690f999-f149-4b9f-9e0e-528b0ab94cdb     Public,
  Calling pac cli inputs: solution import --path D:\a\DevOps-release\DevOps-release\out\solutions\gitactionstesting.zip --async false --import-as-holding false --force-overwrite true --publish-changes true --skip-dependency-check false --convert-to-managed false --max-async-wait-time 60 --activate-plugins true
  Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f149-4b9f-9e0e-528b0ab94cdb
  
  Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.[25](https://github.com/konecorp/DevOps-release/actions/runs/11247885375/job/31272070442#step:4:26).2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  
  
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]
  
    --path                      Path to solution zip file. If not specified, assumes the current folder is a cdsproj project. (alias: -p)
    --activate-plugins          Activate plug-ins and workflows on the solution. (alias: -ap)
    --force-overwrite           Force an overwrite of unmanaged customizations (alias: -f)
    --skip-dependency-check     Skip dependency check against dependencies flagged as product update (alias: -s)
    --import-as-holding         Import the solution as a holding solution. (alias: -h)
    --publish-changes           Publish your changes upon a successful import. (alias: -pc)
    --convert-to-managed        Convert to a managed solution. (alias: -cm)
    --async                     Imports the solution asynchronously. (alias: -a)
    --max-async-wait-time       Max asynchronous wait time in minutes. The default value is 60 minutes. (alias: -wt)
    --settings-file             The .json file with the deployment settings for connection references and environment variables.
    --skip-lower-version        Skip solution import if same or higher version is present in current environment. (alias: -slv)
  
  Error: error: 1
  Error: error: 1
  Error: failed: Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f149-4b9f-9e0e-5[28](https://github.com/konecorp/DevOps-release/actions/runs/11247885375/job/31272070442#step:4:29)b0ab94cdb
  
  Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  
  
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]
  
    --path                      Path to solution zip file. If not specified, assumes the current folder is a cdsproj project. (alias: -p)
    --activate-plugins          Activate plug-ins and workflows on the solution. (alias: -ap)
    --force-overwrite           Force an overwrite of unmanaged customizations (alias: -f)
    --skip-dependency-check     Skip dependency check against dependencies flagged as product update (alias: -s)
    --import-as-holding         Import the solution as a holding solution. (alias: -h)
    --publish-changes           Publish your changes upon a successful import. (alias: -pc)
    --convert-to-managed        Convert to a managed solution. (alias: -cm)
    --async                     Imports the solution asynchronously. (alias: -a)
    --max-async-wait-time       Max asynchronous wait time in minutes. The default value is 60 minutes. (alias: -wt)
    --settings-file             The .json file with the deployment settings for connection references and environment variables.
    --skip-lower-version        Skip solution import if same or higher version is present in current environment. (alias: -slv)
  
  Error: failed: Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f149-4b9f-9e0e-528b0ab94cdb
  
  Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  
  
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]
  
    --path                      Path to solution zip file. If not specified, assumes the current folder is a cdsproj project. (alias: -p)
    --activate-plugins          Activate plug-ins and workflows on the solution. (alias: -ap)
    --force-overwrite           Force an overwrite of unmanaged customizations (alias: -f)
    --skip-dependency-check     Skip dependency check against dependencies flagged as product update (alias: -s)
    --import-as-holding         Import the solution as a holding solution. (alias: -h)
    --publish-changes           Publish your changes upon a successful import. (alias: -pc)
    --convert-to-managed        Convert to a managed solution. (alias: -cm)
    --async                     Imports the solution asynchronously. (alias: -a)
    --max-async-wait-time       Max asynchronous wait time in minutes. The default value is 60 minutes. (alias: -wt)
    --settings-file             The .json file with the deployment settings for connection references and environment variables.
    --skip-lower-version        Skip solution import if same or higher version is present in current environment. (alias: -slv)
  
  Authentication profiles and token cache removed
  The Clear Authentication Result: Authentication profiles and token cache removed
  Error: failed: Error: Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f1[49](https://github.com/konecorp/DevOps-release/actions/runs/11247885375/job/31272070442#step:4:50)-4b9f-9e0e-528b0ab94cdb
  
  Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  
  
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]
  
    --path                      Path to solution zip file. If not specified, assumes the current folder is a cdsproj project. (alias: -p)
    --activate-plugins          Activate plug-ins and workflows on the solution. (alias: -ap)
    --force-overwrite           Force an overwrite of unmanaged customizations (alias: -f)
    --skip-dependency-check     Skip dependency check against dependencies flagged as product update (alias: -s)
    --import-as-holding         Import the solution as a holding solution. (alias: -h)
    --publish-changes           Publish your changes upon a successful import. (alias: -pc)
    --convert-to-managed        Convert to a managed solution. (alias: -cm)
    --async                     Imports the solution asynchronously. (alias: -a)
    --max-async-wait-time       Max asynchronous wait time in minutes. The default value is 60 minutes. (alias: -wt)
    --settings-file             The .json file with the deployment settings for connection references and environment variables.
    --skip-lower-version        Skip solution import if same or higher version is present in current environment. (alias: -slv)
  
  Error: failed: Error: Connected to... EnterpriseAutomation_TCS_QA
  Connected as 5690f999-f149-4b9f-9e0e-[52](https://github.com/konecorp/DevOps-release/actions/runs/11247885375/job/31272070442#step:4:53)8b0ab94cdb
  
  Solution Importing...
  Microsoft PowerPlatform CLI
  Version: 1.25.2+g1c4a1b8
  Online documentation: https://aka.ms/PowerPlatformCLI
  Feedback, Suggestions, Issues: https://github.com/microsoft/powerplatform-build-tools/discussions
  
  Error: CanvasApp import: FAILURE: No file with that name found
      CanvasApp import: FAILURE: No file with that name found
  
  
  Usage: pac solution import [--path] [--activate-plugins] [--force-overwrite] [--skip-dependency-check] [--import-as-holding] [--publish-changes] [--convert-to-managed] [--async] [--max-async-wait-time] [--settings-file] [--skip-lower-version]
  
    --path                      Path to solution zip file. If not specified, assumes the current folder is a cdsproj project. (alias: -p)
    --activate-plugins          Activate plug-ins and workflows on the solution. (alias: -ap)
    --force-overwrite           Force an overwrite of unmanaged customizations (alias: -f)
    --skip-dependency-check     Skip dependency check against dependencies flagged as product update (alias: -s)
    --import-as-holding         Import the solution as a holding solution. (alias: -h)
    --publish-changes           Publish your changes upon a successful import. (alias: -pc)
    --convert-to-managed        Convert to a managed solution. (alias: -cm)
    --async                     Imports the solution asynchronously. (alias: -a)
    --max-async-wait-time       Max asynchronous wait time in minutes. The default value is [60](https://github.com/konecorp/DevOps-release/actions/runs/11247885375/job/31272070442#step:4:61) minutes. (alias: -wt)
    --settings-file             The .json file with the deployment settings for connection references and environment variables.
    --skip-lower-version        Skip solution import if same or higher version is present in current environment. (alias: -slv)
